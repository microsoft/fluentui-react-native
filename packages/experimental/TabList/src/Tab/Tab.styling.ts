import React from 'react';
import { Platform } from 'react-native';

import type { IViewProps } from '@fluentui-react-native/adapters';
import { borderStyles, fontStyles } from '@fluentui-react-native/framework';
import type { Theme } from '@fluentui-react-native/framework';
import type { IconPropsV1 as IconProps } from '@fluentui-react-native/icon';
import type { PressablePropsExtended } from '@fluentui-react-native/interactive-hooks';
import type { TextProps } from '@fluentui-react-native/text';

import type { TabProps, TabSlotProps, TabTokens } from './Tab.types';
import type { TabListState } from '../TabList/TabList.types';

/**
 * Hook to get the style props for each Tab slot.
 */
export const useTabSlotProps = (props: TabProps, tokens: TabTokens, theme: Theme, context: TabListState): TabSlotProps => {
  const { canShowAnimatedIndicator, selectedKey, vertical } = context;

  // Get each slot's props using our final tokens
  const root = React.useMemo<PressablePropsExtended>(
    () => ({
      style: {
        display: 'flex',
        alignItems: 'center',
        flexDirection: tokens.flexDirection,
        alignSelf: 'flex-start',
        justifyContent: 'center',
        padding: 1,
        backgroundColor: tokens.backgroundColor,
        ...(!vertical ? Platform.select({ macos: {}, default: { height: '100%' } }) : {}),
        ...borderStyles.from(tokens, theme),
      },
    }),
    [tokens, theme, vertical],
  );

  const contentContainer = React.useMemo<IViewProps>(
    () => ({
      style: {
        flexDirection: 'row',
        paddingStart: tokens.contentMarginStart,
        paddingEnd: tokens.contentMarginEnd,
      },
    }),
    [tokens.contentMarginStart, tokens.contentMarginEnd],
  );

  const content = React.useMemo<TextProps>(
    () => ({
      style: {
        color: tokens.color,
        ...fontStyles.from(tokens, theme),
      },
    }),
    [tokens, theme],
  );

  const icon = React.useMemo<IconProps>(() => {
    if (props.icon) {
      return { ...props.icon, color: tokens.iconColor, size: tokens.iconSize };
    }
    return {};
  }, [props.icon, tokens.iconColor, tokens.iconSize]);

  const stack = React.useMemo<IViewProps>(
    () => ({
      style: {
        display: 'flex',
        alignItems: 'center',
        flexDirection: 'row',
        alignSelf: 'flex-start',
        justifyContent: 'center',
        marginHorizontal: tokens.stackMarginHorizontal,
        marginVertical: tokens.stackMarginVertical,
      },
    }),
    [tokens.stackMarginHorizontal, tokens.stackMarginVertical],
  );

  const indicatorContainer = React.useMemo<IViewProps>(
    () => ({
      style: {
        backgroundColor: theme.colors.transparentBackground,
        ...(vertical
          ? {
              height: '100%',
              width: tokens.indicatorThickness,
              paddingVertical: tokens.indicatorMargin,
            }
          : {
              width: '100%',
              height: tokens.indicatorThickness,
              paddingHorizontal: tokens.indicatorMargin,
            }),
      },
    }),
    [vertical, tokens.indicatorThickness, tokens.indicatorMargin, theme],
  );

  const indicator = React.useMemo<IViewProps>(
    // if we're the selected tab and we've generated styles for the animated indicator, render the static tab indicator as transparent.
    // The animated indicator will receive styling instead via useTabAnimation hook.
    () => {
      const hideStaticIndicator = props.tabKey === selectedKey && canShowAnimatedIndicator;
      return {
        style: {
          flex: 1,
          borderRadius: tokens.indicatorRadius,
          backgroundColor: hideStaticIndicator ? theme.colors.transparentStroke : tokens.indicatorColor,
        },
      };
    },
    [canShowAnimatedIndicator, props.tabKey, selectedKey, tokens.indicatorColor, tokens.indicatorRadius, theme],
  );

  return { root, contentContainer, content, icon, stack, indicatorContainer, indicator };
};
