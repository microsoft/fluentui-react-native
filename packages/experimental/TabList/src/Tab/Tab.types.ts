import type * as React from 'react';
import type { ViewStyle, ColorValue } from 'react-native';

import type { IViewProps } from '@fluentui-react-native/adapters';
import type { IconPropsV1 as IconProps } from '@fluentui-react-native/icon';
import type { IFocusable, PressableState, PressablePropsExtended } from '@fluentui-react-native/interactive-hooks';
import type { TextProps } from '@fluentui-react-native/text';
import type {
  FontTokens,
  IBackgroundColorTokens,
  IBorderTokens,
  IForegroundColorTokens,
  LayoutTokens,
} from '@fluentui-react-native/tokens';

export const tabName = 'Tab';

export interface TabTokens extends FontTokens, IBorderTokens, IForegroundColorTokens, IBackgroundColorTokens, LayoutTokens {
  /**
   * Horizontal start margin of the tab text. If an icon and text are both rendered, this is the margin beween the icon and text.
   */
  contentMarginStart?: number;

  /**
   * Horizontal end margin of the tab text.
   */
  contentMarginEnd?: number;

  /**
   * Controls order and direction of tab content and indicator.
   */
  flexDirection?: ViewStyle['flexDirection'];

  /**
   * The indicator color.
   */
  indicatorColor?: ColorValue;

  /**
   * The horizontal / vertical margin of the indicator, depending on its orientation.
   */
  indicatorMargin?: number;

  /**
   * The direction the indicator is laying.
   */
  indicatorOrientation?: 'horizontal' | 'vertical';

  /**
   * Border radius of the indicator.
   */
  indicatorRadius?: number;

  /**
   * Thickness of the indicator line.
   */
  indicatorThickness?: number;

  /**
   * The icon color.
   */
  iconColor?: string;

  /**
   * The size of the icon.
   */
  iconSize?: number;

  /**
   * Horizontal margin of the icon + content.
   */
  stackMarginHorizontal?: number;

  /**
   * Vertical margin of the icon + content.
   */
  stackMarginVertical?: number;

  /**
   * States that can be applied to a button
   */
  small?: TabTokens;
  medium?: TabTokens;
  large?: TabTokens;
  vertical?: TabTokens;
  transparent?: TabTokens;
  subtle?: TabTokens;
  hovered?: TabTokens;
  focused?: TabTokens;
  pressed?: TabTokens;
  disabled?: TabTokens;
  selected?: TabTokens;
  hasIcon?: TabTokens;
}

export interface TabProps extends Omit<PressablePropsExtended, 'onPress'> {
  /**
   * A unique key-identifier for each option
   */
  tabKey: string;

  /**
   * Whether or not the tab is selectable
   */
  disabled?: boolean;

  /**
   * Source URL or name of the icon to show on the Button.
   */
  icon?: IconProps;

  /**
   * A RefObject to access the IButton interface. Use this to access the public methods and properties of the component.
   */
  componentRef?: React.RefObject<IFocusable>;

  testID?: string;
}

export interface TabState extends PressableState {
  selected?: boolean;
}

export interface TabInfo {
  props: TabProps & React.ComponentPropsWithRef<any>;
  state: TabState;
}

export interface TabSlotProps {
  root: React.PropsWithRef<PressablePropsExtended>;
  icon: IconProps;
  stack: IViewProps;
  indicatorContainer: IViewProps;
  indicator: IViewProps;
  contentContainer: IViewProps;
  content: TextProps;
}

export interface TabType {
  props: TabProps;
  tokens: TabTokens;
  slotProps: TabSlotProps;
  state: TabState;
}
