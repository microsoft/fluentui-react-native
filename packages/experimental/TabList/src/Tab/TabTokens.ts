import { buildUseTokens } from '@fluentui-react-native/framework';
import type { Theme } from '@fluentui-react-native/framework';
import type { TokensFromTheme } from '@fluentui-react-native/use-styling';

import type { TabTokens } from '..';

export const tabStates: (keyof TabTokens)[] = [
  'small',
  'medium',
  'large',
  'vertical',
  'hovered',
  'disabled',
  'selected',
  'focused',
  'pressed',
  'transparent',
  'subtle',
  'hasIcon',
];

export const defaultTabTokens: TokensFromTheme<TabTokens, Theme> = (t: Theme) =>
  ({
    indicatorOrientation: 'horizontal',
    indicatorThickness: 2,
    borderWidth: 2,
    borderRadius: 4,
    contentMarginStart: 2,
    contentMarginEnd: 2,
    flexDirection: 'column',
    borderColor: t.colors.transparentStroke,
    color: t.colors.neutralForeground2,
    iconColor: t.colors.neutralForeground2,
    indicatorColor: t.colors.transparentStroke,
    indicatorRadius: 99,
    small: {
      iconSize: 20,
      iconMargin: 2,
      indicatorMargin: 8,
      stackMarginHorizontal: 6,
      stackMarginVertical: 8,
      variant: 'body1',
      selected: {
        variant: 'body1Strong',
      },
    },
    medium: {
      iconSize: 20,
      iconMargin: 6,
      indicatorMargin: 12,
      stackMarginHorizontal: 10,
      stackMarginVertical: 12,
      variant: 'body1',
      selected: {
        variant: 'body1Strong',
      },
    },
    large: {
      iconSize: 24,
      iconMargin: 6,
      indicatorMargin: 12,
      stackMarginHorizontal: 10,
      stackMarginVertical: 16,
      variant: 'body2',
      selected: {
        variant: 'subtitle2',
      },
    },
    vertical: {
      flexDirection: 'row-reverse',
      indicatorOrientation: 'vertical',
      small: {
        indicatorMargin: 4,
        stackMarginHorizontal: 6,
        stackMarginVertical: 2,
      },
      medium: {
        indicatorMargin: 8,
        stackMarginHorizontal: 10,
        stackMarginVertical: 6,
      },
      large: {
        indicatorMargin: 10,
        stackMarginHorizontal: 10,
        stackMarginVertical: 8,
      },
    },
    transparent: {
      backgroundColor: t.colors.transparentBackground,
    },
    subtle: {
      backgroundColor: t.colors.subtleBackground,
    },
    selected: {
      color: t.colors.neutralForeground1,
      iconColor: t.colors.compoundBrandForeground1,
      indicatorColor: t.colors.compoundBrandStroke1,
      pressed: {
        color: t.colors.neutralForeground1Pressed,
        iconColor: t.colors.compoundBrandForeground1Pressed,
        indicatorColor: t.colors.compoundBrandStroke1Pressed,
      },
    },
    disabled: {
      color: t.colors.neutralForegroundDisabled,
      iconColor: t.colors.neutralForegroundDisabled,
      selected: {
        color: t.colors.neutralForegroundDisabled,
        iconColor: t.colors.neutralForegroundDisabled,
        indicatorColor: t.colors.neutralForegroundDisabled,
      },
    },
    hovered: {
      color: t.colors.neutralForeground2Hover,
      iconColor: t.colors.neutralForeground2Hover,
      indicatorColor: t.colors.neutralStroke1Hover,
      selected: {
        color: t.colors.neutralForeground1Hover,
        iconColor: t.colors.compoundBrandForeground1Hover,
        indicatorColor: t.colors.compoundBrandStroke1Hover,
      },
      disabled: {
        indicatorColor: t.colors.transparentStroke,
      },
      transparent: {
        backgroundColor: t.colors.transparentBackgroundHover,
      },
      subtle: {
        backgroundColor: t.colors.subtleBackgroundHover,
      },
    },
    pressed: {
      color: t.colors.neutralForeground2Pressed,
      iconColor: t.colors.neutralForeground2Pressed,
      indicatorColor: t.colors.neutralStroke1Pressed,
      transparent: {
        backgroundColor: t.colors.transparentBackgroundPressed,
      },
      subtle: {
        backgroundColor: t.colors.subtleBackgroundPressed,
      },
    },
    focused: {
      borderColor: t.colors.neutralForeground1,
    },
    hasIcon: {
      contentMarginStart: 8,
    },
  } as TabTokens);

export const useTabTokens = buildUseTokens(defaultTabTokens);
