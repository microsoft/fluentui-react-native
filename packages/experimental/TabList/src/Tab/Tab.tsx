/** @jsxRuntime classic */
/** @jsx withSlots */
/** @jsxFrag */

import React from 'react';
import { View, Pressable } from 'react-native';
import type { ViewProps } from 'react-native';

import type { UseTokens } from '@fluentui-react-native/framework';
import { withSlots, compressible, useSlot, useFluentTheme, applyTokenLayers, mergeProps } from '@fluentui-react-native/framework';
import { IconV1 as Icon } from '@fluentui-react-native/icon';
import type { IconPropsV1 as IconProps } from '@fluentui-react-native/icon';
import type { PressablePropsExtended } from '@fluentui-react-native/interactive-hooks';
import type { TextProps } from '@fluentui-react-native/text';
import { Text } from '@fluentui-react-native/text';

import { useTabSlotProps } from './Tab.styling';
import { tabName } from './Tab.types';
import type { TabProps, TabState, TabTokens } from './Tab.types';
import { tabStates, useTabTokens } from './TabTokens';
import { useTab } from './useTab';
import { useTabAnimation } from './useTabAnimation';
import type { TabListState } from '../TabList/TabList.types';
import { TabListContext } from '../TabList/TabListContext';

const tabLookup = (layer: string, state: TabState, props: TabProps, tablistContext: TabListState): boolean => {
  return (
    state[layer] ||
    props[layer] ||
    tablistContext[layer] ||
    layer === tablistContext.appearance ||
    layer === tablistContext.size ||
    (layer === 'hasIcon' && props.icon)
  );
};

export const Tab = compressible<TabProps, TabTokens>((props: TabProps, useTokens: UseTokens<TabTokens>) => {
  const tablist = React.useContext(TabListContext);
  const tab = useTab(props);

  const theme = useFluentTheme();
  let [tokens, cache] = useTokens(theme);

  // Calculate component states to get the correct tokens here (this happens in useSlots for compose components)
  [tokens, cache] = applyTokenLayers(tokens, tabStates, cache, (layer) => tabLookup(layer, tab.state, tab.props, tablist));

  // Get styling props for each Tab slot
  const slotProps = useTabSlotProps(tab.props, tokens, theme, tablist);

  const rootProps = useTabAnimation(props, tablist, tokens, slotProps.root);

  const RootSlot = useSlot<PressablePropsExtended>(Pressable, rootProps);
  const StackSlot = useSlot<ViewProps>(View, slotProps.stack as ViewProps);
  const IndicatorContainerSlot = useSlot<ViewProps>(View, slotProps.indicatorContainer as ViewProps);
  const IndicatorSlot = useSlot<ViewProps>(View, slotProps.indicator as ViewProps);
  const ContentContainerSlot = useSlot<ViewProps>(View, slotProps.contentContainer as ViewProps);
  const ContentSlot = useSlot<TextProps>(Text, slotProps.content);
  const IconSlot = useSlot<IconProps>(Icon, slotProps.icon);

  return (final: TabProps, ...children: React.ReactNode[]) => {
    if (!tab.state) {
      return null;
    }

    // Get label for Tab to use if there's no accessibilityLabel prop passed in.
    let label = '';
    let hasChildren = false;
    React.Children.forEach(children, (child) => {
      if (child !== null) {
        hasChildren = true;
        if (typeof child === 'string') {
          label = child;
        }
      }
    });

    const { icon, tabKey, ...mergedProps } = mergeProps(tab.props, final, {
      accessibilityLabel: tab.props.accessibilityLabel || final.accessibilityLabel || label,
    });

    if (__DEV__ && !hasChildren && !icon) {
      console.warn('A Tab component must render content. Children, an icon, or both should be passed in.');
    }

    return (
      <RootSlot {...mergedProps}>
        <StackSlot>
          {icon && <IconSlot {...icon} />}
          {hasChildren && (
            <ContentContainerSlot>
              {React.Children.map(children, (child, i) =>
                typeof child === 'string' ? (
                  <ContentSlot accessible={false} key={i}>
                    {child}
                  </ContentSlot>
                ) : (
                  child
                ),
              )}
            </ContentContainerSlot>
          )}
        </StackSlot>
        <IndicatorContainerSlot>
          <IndicatorSlot />
        </IndicatorContainerSlot>
      </RootSlot>
    );
  };
}, useTabTokens);
Tab.displayName = tabName;

export default Tab;
