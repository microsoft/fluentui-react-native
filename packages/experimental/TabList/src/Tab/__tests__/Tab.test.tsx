import * as React from 'react';

import { checkReRender } from '@fluentui-react-native/test-tools';
import * as renderer from 'react-test-renderer';

import Tab from '../Tab';

describe('Tab component tests', () => {
  it('Tab default props', () => {
    const tree = renderer.create(<Tab tabKey="1">Tab 1</Tab>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Tab disabled', () => {
    const tree = renderer
      .create(
        <Tab tabKey="1" disabled>
          Tab 1
        </Tab>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Tab render icon only', () => {
    const tree = renderer
      .create(
        <Tab
          icon={{
            fontSource: {
              fontFamily: 'Arial',
              codepoint: 0x2663,
            },
          }}
          tabKey="1"
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Tab render icon + text', () => {
    const tree = renderer
      .create(
        <Tab
          icon={{
            fontSource: {
              fontFamily: 'Arial',
              codepoint: 0x2663,
            },
          }}
          tabKey="1"
        >
          Tab 1
        </Tab>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Customized Tab', () => {
    const CustomTab = Tab.customize({
      variant: 'headerSemibold',
      stackMarginVertical: 16,
      stackMarginHorizontal: 16,
      indicatorThickness: 4,
      color: 'red',
    });
    const tree = renderer.create(<CustomTab tabKey="1">Tab 1</CustomTab>);
    expect(tree).toMatchSnapshot();
  });

  it('Tab re-renders correctly', () => {
    checkReRender(
      () => (
        <Tab tabKey="1" accessibilityPositionInSet={1}>
          Tab 1
        </Tab>
      ),
      2,
    );
  });
});
