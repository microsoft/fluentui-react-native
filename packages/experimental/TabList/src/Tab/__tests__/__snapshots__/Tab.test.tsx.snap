// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tab component tests Customized Tab 1`] = `
<View
  accessibilityActions={
    [
      {
        "name": "Select",
      },
    ]
  }
  accessibilityLabel="Tab 1"
  accessibilityPositionInSet={0}
  accessibilityRole="tab"
  accessibilitySetSize={0}
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "multiselectable": undefined,
      "required": undefined,
      "selected": false,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  focusable={true}
  keyUpEvents={
    [
      {
        "key": " ",
      },
      {
        "key": "Enter",
      },
    ]
  }
  onAccessibilityAction={[Function]}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onLayout={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "alignSelf": "flex-start",
      "backgroundColor": "#00000000",
      "borderColor": "#00000000",
      "borderRadius": 4,
      "borderWidth": 2,
      "display": "flex",
      "flexDirection": "column",
      "height": "100%",
      "justifyContent": "center",
      "padding": 1,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "alignSelf": "flex-start",
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "center",
        "marginHorizontal": 6,
        "marginVertical": 8,
      }
    }
  >
    <View
      style={
        {
          "flexDirection": "row",
          "paddingEnd": 2,
          "paddingStart": 2,
        }
      }
    >
      <Text
        accessible={false}
        style={
          {
            "color": "red",
            "fontFamily": "Segoe UI",
            "fontSize": 12,
            "fontWeight": "400",
            "margin": 0,
          }
        }
      >
        Tab 1
      </Text>
    </View>
  </View>
  <View
    style={
      {
        "backgroundColor": "#00000000",
        "height": 4,
        "paddingHorizontal": 8,
        "width": "100%",
      }
    }
  >
    <View
      style={
        {
          "backgroundColor": "#00000000",
          "borderRadius": 99,
          "flex": 1,
        }
      }
    />
  </View>
</View>
`;

exports[`Tab component tests Tab default props 1`] = `
<View
  accessibilityActions={
    [
      {
        "name": "Select",
      },
    ]
  }
  accessibilityLabel="Tab 1"
  accessibilityPositionInSet={0}
  accessibilityRole="tab"
  accessibilitySetSize={0}
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "multiselectable": undefined,
      "required": undefined,
      "selected": false,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  focusable={true}
  keyUpEvents={
    [
      {
        "key": " ",
      },
      {
        "key": "Enter",
      },
    ]
  }
  onAccessibilityAction={[Function]}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onLayout={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "alignSelf": "flex-start",
      "backgroundColor": "#00000000",
      "borderColor": "#00000000",
      "borderRadius": 4,
      "borderWidth": 2,
      "display": "flex",
      "flexDirection": "column",
      "height": "100%",
      "justifyContent": "center",
      "padding": 1,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "alignSelf": "flex-start",
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "center",
        "marginHorizontal": 6,
        "marginVertical": 8,
      }
    }
  >
    <View
      style={
        {
          "flexDirection": "row",
          "paddingEnd": 2,
          "paddingStart": 2,
        }
      }
    >
      <Text
        accessible={false}
        style={
          {
            "color": "#424242",
            "fontFamily": "Segoe UI",
            "fontSize": 12,
            "fontWeight": "400",
            "margin": 0,
          }
        }
      >
        Tab 1
      </Text>
    </View>
  </View>
  <View
    style={
      {
        "backgroundColor": "#00000000",
        "height": 2,
        "paddingHorizontal": 8,
        "width": "100%",
      }
    }
  >
    <View
      style={
        {
          "backgroundColor": "#00000000",
          "borderRadius": 99,
          "flex": 1,
        }
      }
    />
  </View>
</View>
`;

exports[`Tab component tests Tab disabled 1`] = `
<View
  accessibilityActions={
    [
      {
        "name": "Select",
      },
    ]
  }
  accessibilityLabel="Tab 1"
  accessibilityPositionInSet={0}
  accessibilityRole="tab"
  accessibilitySetSize={0}
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": true,
      "expanded": undefined,
      "multiselectable": undefined,
      "required": undefined,
      "selected": false,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  focusable={false}
  keyUpEvents={
    [
      {
        "key": " ",
      },
      {
        "key": "Enter",
      },
    ]
  }
  onAccessibilityAction={[Function]}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onLayout={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "alignSelf": "flex-start",
      "backgroundColor": "#00000000",
      "borderColor": "#00000000",
      "borderRadius": 4,
      "borderWidth": 2,
      "display": "flex",
      "flexDirection": "column",
      "height": "100%",
      "justifyContent": "center",
      "padding": 1,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "alignSelf": "flex-start",
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "center",
        "marginHorizontal": 6,
        "marginVertical": 8,
      }
    }
  >
    <View
      style={
        {
          "flexDirection": "row",
          "paddingEnd": 2,
          "paddingStart": 2,
        }
      }
    >
      <Text
        accessible={false}
        style={
          {
            "color": "#bdbdbd",
            "fontFamily": "Segoe UI",
            "fontSize": 12,
            "fontWeight": "400",
            "margin": 0,
          }
        }
      >
        Tab 1
      </Text>
    </View>
  </View>
  <View
    style={
      {
        "backgroundColor": "#00000000",
        "height": 2,
        "paddingHorizontal": 8,
        "width": "100%",
      }
    }
  >
    <View
      style={
        {
          "backgroundColor": "#00000000",
          "borderRadius": 99,
          "flex": 1,
        }
      }
    />
  </View>
</View>
`;

exports[`Tab component tests Tab render icon + text 1`] = `
<View
  accessibilityActions={
    [
      {
        "name": "Select",
      },
    ]
  }
  accessibilityLabel="Tab 1"
  accessibilityPositionInSet={0}
  accessibilityRole="tab"
  accessibilitySetSize={0}
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "multiselectable": undefined,
      "required": undefined,
      "selected": false,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  focusable={true}
  keyUpEvents={
    [
      {
        "key": " ",
      },
      {
        "key": "Enter",
      },
    ]
  }
  onAccessibilityAction={[Function]}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onLayout={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "alignSelf": "flex-start",
      "backgroundColor": "#00000000",
      "borderColor": "#00000000",
      "borderRadius": 4,
      "borderWidth": 2,
      "display": "flex",
      "flexDirection": "column",
      "height": "100%",
      "justifyContent": "center",
      "padding": 1,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "alignSelf": "flex-start",
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "center",
        "marginHorizontal": 6,
        "marginVertical": 8,
      }
    }
  >
    <Text
      accessible={true}
      style={
        {
          "color": "#424242",
          "fontFamily": "Arial",
          "fontSize": 20,
        }
      }
    >
      â™£
    </Text>
    <View
      style={
        {
          "flexDirection": "row",
          "paddingEnd": 2,
          "paddingStart": 8,
        }
      }
    >
      <Text
        accessible={false}
        style={
          {
            "color": "#424242",
            "fontFamily": "Segoe UI",
            "fontSize": 12,
            "fontWeight": "400",
            "margin": 0,
          }
        }
      >
        Tab 1
      </Text>
    </View>
  </View>
  <View
    style={
      {
        "backgroundColor": "#00000000",
        "height": 2,
        "paddingHorizontal": 8,
        "width": "100%",
      }
    }
  >
    <View
      style={
        {
          "backgroundColor": "#00000000",
          "borderRadius": 99,
          "flex": 1,
        }
      }
    />
  </View>
</View>
`;

exports[`Tab component tests Tab render icon only 1`] = `
<View
  accessibilityActions={
    [
      {
        "name": "Select",
      },
    ]
  }
  accessibilityLabel=""
  accessibilityPositionInSet={0}
  accessibilityRole="tab"
  accessibilitySetSize={0}
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "multiselectable": undefined,
      "required": undefined,
      "selected": false,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  focusable={true}
  keyUpEvents={
    [
      {
        "key": " ",
      },
      {
        "key": "Enter",
      },
    ]
  }
  onAccessibilityAction={[Function]}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onLayout={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "alignSelf": "flex-start",
      "backgroundColor": "#00000000",
      "borderColor": "#00000000",
      "borderRadius": 4,
      "borderWidth": 2,
      "display": "flex",
      "flexDirection": "column",
      "height": "100%",
      "justifyContent": "center",
      "padding": 1,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "alignSelf": "flex-start",
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "center",
        "marginHorizontal": 6,
        "marginVertical": 8,
      }
    }
  >
    <Text
      accessible={true}
      style={
        {
          "color": "#424242",
          "fontFamily": "Arial",
          "fontSize": 20,
        }
      }
    >
      â™£
    </Text>
  </View>
  <View
    style={
      {
        "backgroundColor": "#00000000",
        "height": 2,
        "paddingHorizontal": 8,
        "width": "100%",
      }
    }
  >
    <View
      style={
        {
          "backgroundColor": "#00000000",
          "borderRadius": 99,
          "flex": 1,
        }
      }
    />
  </View>
</View>
`;
